tests/test_sparse.py

```diff
@@ -23,81 +23,74 @@ def rng():
 
 
 @pytest.mark.parametrize("dtype", [np.int64, np.float64, np.complex128])
-@pytest.mark.parametrize("order", ["C", "F"])
+@pytest.mark.parametrize("order", ["C", "F", None])
 def test_wrappers(dtype, order):
     A = np.array([[0, 0, 4], [1, 0, 0], [2, 0, 5], [3, 0, 0]], dtype=dtype)
-    B = np.stack([A, A], axis=2, dtype=dtype)
-    scalar = finch.Tensor(finch.Element(2), np.array(2), order=order)
+    B = np.array(np.stack([A, A], axis=2, dtype=dtype))
 
-    levels = finch.Dense(finch.SparseList(finch.SparseList(finch.Element(dtype(0.0)))))
-    B_finch = finch.Tensor(levels, B, order=order)
+    storage = finch.Storage(
+        finch.Dense(finch.SparseList(finch.SparseList(finch.Element(dtype(0.0))))), order=order
+    )
+    B_finch = finch.Tensor(B, order=order).to_device(storage)
 
     assert_equal(B_finch.todense(), B)
-    #assert_equal((B_finch * scalar + B_finch).todense(), B * 2 + B)
 
-    levels = finch.Dense(finch.Dense(finch.Element(dtype(1.0))))
-    A_finch = finch.Tensor(levels, A, order=order)
+    storage = finch.Storage(
+        finch.Dense(finch.Dense(finch.Element(dtype(1.0)))), order=order
+    )
+    A_finch = finch.Tensor(A, order=order).to_device(storage)
 
     assert_equal(A_finch.todense(), A)
-    #assert_equal((A_finch * scalar + A_finch).todense(), A * 2 + A)
-
     assert A_finch.todense().dtype == A.dtype and B_finch.todense().dtype == B.dtype
 
 
-@pytest.mark.skip(reason="a copy is made")
-@pytest.mark.parametrize("dtype", [np.int64])
-@pytest.mark.parametrize("order", ["C", "F"])
+@pytest.mark.parametrize("dtype", [np.int64, np.float64, np.complex128])
+@pytest.mark.parametrize("order", ["C", "F", None])
 def test_no_copy_fully_dense(dtype, order, arr3d):
+    arr = np.array(arr3d, dtype=dtype, order=order)
 
-    levels = finch.Dense(finch.Dense(finch.Dense(finch.Element(dtype(0.0)))))
-    arr_finch = finch.Tensor(levels, arr3d, order=order)
-
+    arr_finch = finch.Tensor(arr, order=order)
     arr_todense = arr_finch.todense()
 
-    assert np.shares_memory(arr3d, arr_todense)
+    assert_equal(arr_todense, arr)
+    assert np.shares_memory(arr_todense, arr)
 
 
 def test_coo(rng):
-    coords = np.asarray(
-        [[0, 1, 2, 3, 4], [0, 1, 2, 3, 4]],
-        dtype=np.intp,
+    coords = (
+        np.asarray([0, 1, 2, 3, 4], dtype=np.intp),
+        np.asarray([0, 1, 2, 3, 4], dtype=np.intp),
     )
     data = rng.random(5)
-    scalar = finch.Tensor(finch.Element(2), np.array(2))
 
-    arr_pydata = sparse.COO(coords, data, shape=(5, 5))
+    arr_pydata = sparse.COO(np.vstack(coords), data, shape=(5, 5))
     arr = arr_pydata.todense()
-    arr_finch = finch.COO(coords, data, shape=(5, 5))
+    arr_finch = finch.Tensor.construct_coo(coords, data, shape=(5, 5))
```

Comment:
Could we define an alias to make this invocation simpler? i.e. `finch.COO` vs. `finch.Tensor.construct_coo`, so that `construct_coo` is more of an internal syntax?

Reply:
Makes sense - let me address it in the follow-up PR!